include/nbl/builtin/hlsl/emulated_float64_t.hlsl

```diff
@@ -0,0 +1,118 @@
+
+using float32_t = float;
+//using emulated_float64_t = double;
+
+namespace emulated
+{
+    struct emulated_float64_t
+    {
+        // TODO: change to `uint64_t` when on the emulation stage
+        using storage_t = float32_t;
+
+        storage_t data;
+
+        // constructors
+        // TODO: specializations?
+        template <typename T>
+        static emulated_float64_t create(T val)
+        { 
+            emulated_float64_t output;
+            output.data = val;
+            return output;
+        }
+
+        // arithmetic operators
+        emulated_float64_t operator+(const emulated_float64_t rhs)
+        {
+            emulated_float64_t retval;
+            retval.data = data + rhs.data;
+            return retval;
+        }
+
+        emulated_float64_t operator-(const emulated_float64_t rhs)
+        {
+            emulated_float64_t retval;
+            retval.data = data - rhs.data;
+            return retval;
+        }
+
+        emulated_float64_t operator*(const emulated_float64_t rhs)
+        {
+            emulated_float64_t retval;
+            retval.data = data * rhs.data;
+            return retval;
+        }
+
+        emulated_float64_t operator/(const emulated_float64_t rhs)
+        {
+            emulated_float64_t retval;
+            retval.data = data / rhs.data;
+            return retval;
+        }
+
+        // relational operators
+        bool operator==(const emulated_float64_t rhs) { return !(uint64_t(data) ^ uint64_t(rhs.data)); }
+        bool operator!=(const emulated_float64_t rhs) { return uint64_t(data) ^ uint64_t(rhs.data); }
+        bool operator<(const emulated_float64_t rhs) { return data < rhs.data; }
+        bool operator>(const emulated_float64_t rhs) { return data > rhs.data; }
+        bool operator<=(const emulated_float64_t rhs) { return !operator>(rhs); }
+        bool operator>=(const emulated_float64_t rhs) { return !operator<(rhs); }
+
+        //logical operators
+        bool operator&&(const emulated_float64_t rhs) { return bool(data) && bool(rhs.data); }
+        bool operator||(const emulated_float64_t rhs) { return bool(data) || bool(rhs.data); }
+        bool operator!() { return !bool(data); }
+
+        // conversion operators
+        operator bool() { return bool(data); }
+        operator int() { return int(data); }
+        operator uint32_t() { return uint32_t(data); }
+        operator uint64_t() { return uint64_t(data); }
+        operator float() { return float(data); }

```

Comment:
imo, none of these will work

Reply:
it works according to my tests

Reply:
mind linking a repro?

Reply:
triple and quadruple checked with DXC, they don't really.